features/authentication/authentication.api.ts

```diff
@@ -0,0 +1,71 @@
+import { Platform } from "react-native";
+import * as Device from "expo-device";
+import { api } from "@/utils/api/api";
+import { z } from "zod";
+
+const deviceOSEnum = z.enum(["android", "ios", "web"]);
+
+const createUserResponseSchema = z.object({
+  id: z.string(),
+  privyUserId: z.string(),
+  device: z.object({
+    id: z.string(),
+    os: deviceOSEnum,
+    name: z.string().nullable(),
+  }),
+  identity: z.object({
+    id: z.string(),
+    privyAddress: z.string(),
+    xmtpId: z.string().nullable(),
+  }),
+  profile: z
+    .object({
+      id: z.string(),
+      name: z.string(),
+      description: z.string().nullable(),
+    })
+    .nullable(),
+});
+
+type CreateUserResponse = z.infer<typeof createUserResponseSchema>;
+
+export const createUser = async (args: {
+  privyUserId: string;
+  smartContractWalletAddress: string;
+  inboxId: string;
+}): Promise<CreateUserResponse> => {
+  const { privyUserId, smartContractWalletAddress, inboxId } = args;
+
+  const requestData = {
+    privyUserId,
+    device: {
+      os: Platform.OS.toLowerCase(),
+      name: Device.modelId,
+    },
+    identity: {
+      privyAddress: smartContractWalletAddress,
+      xmtpId: inboxId,
+    },
+  };
+
+  const response = await api.post<CreateUserResponse>(
+    "/api/v1/users",
+    requestData
+  );
+  return createUserResponseSchema.parse(response.data);
```

Comment:
Wonder if we should do `safeParse` here. Just because let's say the Backend accitdently change 1 property and so this request no longer work but in reality it would work for 99% of the time.

Reply:
why are we doing client side validation of the response data at all? we own the backend and know exactly what it's going to return. seems like a lot of extra work when we really just need a type assertion.

Reply:
just as a sanity check. not much extra work to define the type and parse with zod, so we just do it. 

even though we own it, it can still change and we want to always catch errors as early as possible, which would be at the repsonse site rather than downstream

Reply:
I think we just want to parse everything and keep it up to date. no sense in letting these get out of date. 

later @rygine had a really good idea about just building an sdk for the backend, so not going to bike shed too much ont hisatm